### PR TITLE
[cuDNN SDPA] add sm86/sm89 flash attention support

### PR DESCRIPTION
**Pull request by Cjkkkk**
remove sm80/sm90 only constraint to include sm86/sm89 as well since non flash version attention is removed from cuDNN SDPA and this constraint can be relaxed.

### CODE REVIEW

**Comment by Cjkkkk** [comment]
@superbobry could you help review this? related discussion: https://github.com/google/jax/issues/22546

File: jax/_src/cudnn/fused_attention_stablehlo.py [hunk 1]
```diff
@@ -986,8 +979,8 @@ def dot_product_attention(query: Array,
   """
   # check if cuDNN is installed
   cudnn_version = check_cudnn_version()
-  # only support Ampere and Hopper for now
-  check_compute_capability((80, 90))
+  # only support at least Ampere
+  jtu.is_cuda_compute_capability_at_least("8.0")
```
Comment (superbobry) [inline jax/_src/cudnn/fused_attention_stablehlo.py:983]: `jtu` is for testing. Perhaps inline it here instead?

Note also that atm you aren't asserting anything.

File: tests/fused_attention_stablehlo_test.py [hunk 2]
```diff
@@ -159,7 +158,7 @@ def setUp(self):
       self.skipTest("Requires more than 4 devices.")
     try:
       cudnn_version = check_cudnn_version()
-      check_compute_capability((80, 90))
+      jtu.is_cuda_compute_capability_at_least("8.0")
```
Comment (superbobry) [inline tests/fused_attention_stablehlo_test.py:161]:
```
if not jtu.is_cuda_compute_capability_at_least("8.0"):
  self.skipTest(...)
```

File: tests/fused_attention_stablehlo_test.py [hunk 3]
```diff
@@ -340,7 +339,7 @@ def test_sdpa_var_seq(self):
   def test_sdpa_broadcast_bias_and_dbias(self):
     try:
       cudnn_version = check_cudnn_version()
-      check_compute_capability((90,))
+      jtu.is_cuda_compute_capability_at_least("9.0")
```
Comment (superbobry) [inline tests/fused_attention_stablehlo_test.py:342]: As above.

**Review by superbobry — CHANGES_REQUESTED** [review]

**Comment by Cjkkkk** [comment]
@superbobry all comments addressed and rebased. Could you take an another look? Thanks!

**Comment by Cjkkkk** [comment]
Rebased again to support blackwell: https://github.com/google/jax/pull/22608

**Review by superbobry — APPROVED** [review]

**Comment by MasterSkepticista** [comment]
Apologies if this isn't the relevant PR to ask: Is there a plan to add `dropout` capability on `attn_weights` in the SDPA API?

**Pull request merged**